src/main/java/com/udojava/evalex/LazyIfNumber.java

```diff
@@ -0,0 +1,33 @@
+package com.udojava.evalex;
+
+import com.udojava.evalex.Expression.LazyNumber;
+import java.math.BigDecimal;
+import java.util.List;
+
+public class LazyIfNumber implements LazyNumber {
+
+  List<LazyNumber> lazyParams;
+
+  public LazyIfNumber(List<LazyNumber> lazyParams) {
+    this.lazyParams = lazyParams;
+  }
+
+  @Override
+  public BigDecimal eval() {
+    BigDecimal result = lazyParams.get(0).eval();
+    assertNotNull(result);
+    boolean isTrue = result.compareTo(BigDecimal.ZERO) != 0;
+    return isTrue ? lazyParams.get(1).eval() : lazyParams.get(2).eval();
+  }
+
+  @Override
+  public String getString() {
+    return "IF";
```

Comment:
I believe that other implementations are either returning an actual `String` representation of the operation, or the result as `String`. Could you double check that?